rflx/pyrflx/typevalue.py

```diff
@@ -766,6 +794,109 @@ def _preset_fields(self, fld: str) -> None:
             self._last_field = nxt
             nxt = self._next_field(nxt)
 
+    def set_checksum_function(self, checksum_field_name: str, checksum_method: Callable) -> None:
+        if checksum_field_name not in self.fields:
+            raise KeyError(f"Field {checksum_field_name} is not defined")
+        for field_name, checksum in self._checksums.items():
+            if field_name == checksum_field_name:
+                checksum.set_checksum_function(checksum_method)
+                return
+        raise KeyError(f"Field {checksum_field_name} has not been defined as a checksum field")
+
+    def _is_checksum_settable(self, checksum_aspect: "MessageValue.Checksum") -> bool:
+        def valid_path(value_range: ValueRange) -> bool:
+            expr: Dict[Expr, str] = dict.fromkeys([value_range.lower, value_range.upper])
+
+            for k in expr:
+                if isinstance(k, Sub):
+                    assert isinstance(k.left, (First, Last))
+                    expr[k] = str(k.left.prefix)
+                elif isinstance(k, Add):
+                    for t in k.terms:
+                        if isinstance(t, (First, Last)):
+                            expr[k] = str(t.prefix)
+                else:
+                    assert isinstance(k, (First, Last))
+                    expr[k] = str(k.prefix)
```

Comment:
Why `k`? Wouldn't be `e` more appropriate for an expression?

Reply:
`k` for key, because expr is a dict. Changed it in c238ac0